File: src/17.py
ONES = ['', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine', 'ten', 'eleven', 'twelve', 'thirteen', 'fourteen', 'fifteen']

TENS = ['', '', 'twen', 'thir', 'for', 'fif']

BIGS = ['','','m', 'b', 'tr', 'quadr', 'quint', 'sext', 'sept', 'oct', 'non', 'dec', 'undec', 'duodec', 'tredec', 'quattuordec', 'quindec', 'sexdec', 'septendec', 'octodec', 'novemdec', 'vigint']

def spell_small(i):
    s = ""
    x,y=divmod(i,100)
    s+=f"{ONES[x]} hundred{' and 'if y else''}"if x else''
    s+=ONES[y]if y<16 else f"{ONES[y-10]}teen"if y<20 else''
    x,y=divmod(y,10)

    s+=f"{TENS[x] if x<6 else ONES[x]}ty{'-'if y else''}{ONES[y] if y else''}"if x>1 else''
    return s.replace('tt','t')

def num_builder(i,big,result):
    n,i = divmod(i,1000)
    s = spell_small(i)
    if not i:0
    elif big == 1:s += BIGS[big]+" thousand"
    elif big:s += " "+BIGS[big]+"illion"
    if s:result.insert(0,s)
    if n:num_builder(n,big+1,result)

def spelling_number(i):
    result = []
    if i == 0:result += ["zero"]
    big_counter = 0
    num_builder(i,0,result)
    return ", ".join(result)

S=''
for i in range(1,1001):
	S+=spelling_number(i)
print(len(S)-S.count(' ')-S.count(',')-S.count('-'))

File: src/10-19/17.ml
open Stdio

let ones = [|""; "one"; "two"; "three"; "four"; "five"; "six"; "seven"; "eight"; "nine"|]
let teens = [| "ten"; "eleven"; "twelve"; "thirteen"; "fourteen"; "fifteen"; "sixteen"; "seventeen"; "eighteen"; "nineteen" |]
let tens = [|""; ""; "twenty"; "thirty"; "forty"; "fifty"; "sixty"; "seventy"; "eighty"; "ninety"|]

let rec length_of_number n =
  if n < 10 then String.length ones.(n)
  else if n < 20 then String.length teens.(n - 10)
  else if n < 100 then String.length tens.(n / 10) + length_of_number (n mod 10)
  else if n < 1000 then
    String.length ones.(n / 100) + String.length "hundred" +
    (if n mod 100 <> 0 then String.length "and" + length_of_number (n mod 100) else 0)
  else String.length "onethousand"

let total_length = List.fold_left (fun acc x -> acc + length_of_number x) 0 (List.init 1000 (fun x -> x + 1))

let () = printf "%d\n" total_length
